### PR TITLE
feat: done task 4

### PR DESCRIPTION
**Pull request by Sergey-Churilin**
[Link to FE app](https://d2ykvchm0vckyx.cloudfront.net/). New product can be created from admin panel.
Link to product API: https://1oyyu6n9d2.execute-api.eu-west-1.amazonaws.com/dev
[Link to swagger](https://c6cm2bz2f4.execute-api.eu-west-1.amazonaws.com/swagger)

- [x] - Task 4.1 is implemented
- [x] - Task 4.2 is implemented lambda links are provided and returns data
- [x] - Task 4.3 is implemented lambda links are provided and products is stored in DB (call Task 4.2 to see the product)
- [x] - Your own Frontend application is integrated with Product Service (/products API) and products from Product Service are represented on Frontend. Link to a working Frontend application is provided for cross-check reviewer.

Optional
- [x] +1 (All languages) - POST /products lambda functions returns error 400 status code if product data is invalid
- [x] +1 (All languages) - All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- [x] +1 (All languages) - All lambdas do console.log for each incoming requests and their arguments
- [ ] +1 (All languages) - Use RDS instance instead fo DynamoDB tables
- [x] +1 (All languages) - Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa)

### CODE REVIEW

File: product-service-ts/src/functions/getProductById/handler.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import { formatJSONResponse } from "@libs/api-gateway";
+import { middyfy } from "@libs/lambda";
+import AWS from "aws-sdk";
+import schema from "./schema";
+import { DocumentClient } from "aws-sdk/lib/dynamodb/document_client";
+
+const dynamo: DocumentClient = new AWS.DynamoDB.DocumentClient();
+const queryProduct = async (id) => {
+    const result = await dynamo
+        .query({ TableName: process.env.TABLE_NAME, KeyConditionExpression: "id = :id", ExpressionAttributeValues: { ":id": id } })
```
Comment (Guria) [inline product-service-ts/src/functions/getProductById/handler.ts:11]: you may try to use `get` method to receive single item

File: product-service-ts/src/functions/createProduct/schema.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+export default {
```
Comment (Guria) [inline product-service-ts/src/functions/createProduct/schema.ts:1]: this schema is unused

File: product-service-ts/src/functions/getProductById/schema.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+export default {
```
Comment (Guria) [inline product-service-ts/src/functions/getProductById/schema.ts:1]: this schema is unused

File: product-service-ts/src/functions/getProductsList/schema.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+export default {
```
Comment (Guria) [inline product-service-ts/src/functions/getProductsList/schema.ts:1]: this schema is unused

File: product-service-ts/swagger/functions.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (Guria) [inline product-service-ts/swagger/functions.js]: generated code usually shouldn't be stored in the repo

**Review by Guria — APPROVED** [review]
## Notes
* swagger - no payload for POST `/products`
* there is no checks for negative numbers in price and acount
* `createProduct` endpoint doesn't returns CORS headers when it returns 400 error so browser can't read error message
* there are 2 implementations and generated code for swagger in review, which complicates the process of review
* JSON schemas are defined but not used anywhere

## Evaluation criteria:
:heavy_check_mark: Reviewers should verify the lambda functions by invoking them through provided URLs.

:heavy_check_mark: 1 - Task 4.1 is implemented: there is a script to fill tables with test examples
:question: 3 - Task 4.2 is implemented lambda links are provided and returns data:
  * :heavy_check_mark: [`getProductsList`](https://1oyyu6n9d2.execute-api.eu-west-1.amazonaws.com/dev/products) - works as expected. Newly created item is present in response
  * :heavy_minus_sign: [`getProductById`](https://1oyyu6n9d2.execute-api.eu-west-1.amazonaws.com/dev/products/e1dfdc29-daff-46b5-99fd-3c3f84524a99) - returns 500 error for existing item id

:heavy_check_mark: 4 - Task 4.3 is implemented lambda links are provided and products is stored in DB (call Task 4.2 to see the product)
:heavy_check_mark: 5 - Your own Frontend application is integrated with Product Service (/products API) and products from Product Service are represented on Frontend.

## Additional (optional) tasks
:heavy_check_mark: +1 POST /products lambda functions returns error 400 status code if product data is invalid
:heavy_check_mark: +1 All lambdas return error 500 status code on any error (DB connection, any unhandled error in code) (:exclamation: but it doesn't set CORS headers)
:heavy_check_mark: +1 All lambdas do console.log for each incoming requests and their arguments
:heavy_minus_sign: App doesn't use RDS instance instead fo DynamoDB tables. 
:heavy_check_mark: +1 Transaction based creation of product (in case stock creation is failed then related to this stock product is not created and not ready to be used by the end user and vice versa)

## Evaluation
Technically it is 5. Because `getProductById` is not working. But I still see that you have made lot of efforts. I will give you a 9 since I believe you will fix it quickly.

File: product-service-ts/src/functions/getProductById/handler.ts (hunk 1 above)
Comment (Sergey-Churilin) [reply product-service-ts/src/functions/getProductById/handler.ts:11]: Updated for getstock, for some reason doesn't work for get produce. Will fix later

File: product-service-ts/src/functions/createProduct/schema.ts (hunk 2 above)
Comment (Sergey-Churilin) [reply product-service-ts/src/functions/createProduct/schema.ts:1]: Yep, will fix later, didn't have time to do it now =(

File: product-service-ts/swagger/functions.js (hunk 5 above)
Comment (Sergey-Churilin) [reply product-service-ts/swagger/functions.js]: Add to git ignore

**Pull request merged**